### PR TITLE
[VPU] Change data structure in processTrivialCases method

### PR DESCRIPTION
**Pull request by AndrewBakalinIntel**
Fix the review note from https://github.com/openvinotoolkit/openvino/pull/4060
Change data structure from `std::vector` to `std::pair` to make the intention (select two data objects to compound trivial case) more clear.

### CODE REVIEW

**Comment by AndrewBakalinIntel** [comment]
@gladilov-gleb could you please take a look?

**Comment by gladilov-gleb** [comment]
> @gladilov-gleb could you please take a look?

Name of PR is not descriptive

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 1]
```diff
@@ -379,27 +379,31 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
...
+                VPU_THROW_UNLESS(ieDataToTrivialCase.count(origData) == 0 || ieDataToTrivialCase[origData].second == nullptr,
+                                 "There can't be two data output objects associated with the same original IE data object with name {}",
+                                 origData->getName());
+                ieDataToTrivialCase[origData].second = data;
+            } else {
+                VPU_THROW_UNLESS(ieDataToTrivialCase.count(origData) == 0 || ieDataToTrivialCase[origData].first == nullptr,
+                                 "There can't be two data input/const objects associated with the same original IE data object with name {}",
+                                 origData->getName());
+                ieDataToTrivialCase[origData].first = data;
+            }
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:396]: Nitpick:

```suggestion
            auto& trivialCase = ieDataToTrivialCase[origData];
            auto& destination = data->usage() == DataUsage::Output ? trivialCase.second : trivialCase.first;
            VPU_THROW_UNLESS(ieDataToTrivialCase.count(origData) == 0 || destination == nullptr, 
                "Encountered IE data object {} which has two vpu data objects {} and {} of the same type {} associated with it, while only one is permitted",
                origData->getName(), destination->name(), data->name(), destination->usage());
            destination = data;
```

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 2]
```diff
@@ -379,27 +379,31 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
 }
 
 void FrontEnd::processTrivialCases(const Model& model) {
-    std::unordered_map<ie::DataPtr, DataVector> ieToVpuDataVector;
+    std::unordered_map<ie::DataPtr, std::pair<Data, Data>> ieDataToTrivialCase;
     for (const auto& data : model->datas()) {
         const auto& origData = data->origData();
         if (origData != nullptr) {
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:385]: Nitpick: save a level of indentation

```cpp
if (!condition) {
    continue;
}

// body
```

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp [hunk 3]
```diff
@@ -379,27 +379,31 @@ void FrontEnd::parseLayer(const Model& model, const ie::CNNLayerPtr& layer, cons
...
-    for (const auto& dataObjectsWithTheSameOrigData : ieToVpuDataVector) {
-        if (dataObjectsWithTheSameOrigData.second.size() > 1) {
-            VPU_THROW_UNLESS(dataObjectsWithTheSameOrigData.second.size() == 2,
-                             "There can't be more than two data objects associated with the same original IE data object with name {}",
-                             dataObjectsWithTheSameOrigData.first->getName());
-            trivialCases.push_back(dataObjectsWithTheSameOrigData.second);
-        }
-    }
+    for (const auto& trivialCase : ieDataToTrivialCase) {
+        const auto& unconnectedInput = trivialCase.second.first;
```
Comment (gladilov-gleb) [inline inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp:401]: Nitpick: add variable with descriptive name for `trivialCase.second`. It'll make code more readable - `.second.first` is a bit complex

**Comment by AndrewBakalinIntel** [comment]
> > @gladilov-gleb could you please take a look?
> 
> Name of PR is not descriptive

Thanks, fixed.

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp (hunk 1 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp]: Good suggestion, fixed.

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp (hunk 2 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp]: Thanks! Done.

File: inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp (hunk 3 above)
Comment (AndrewBakalinIntel) [reply inference-engine/src/vpu/graph_transformer/src/frontend/frontend.cpp]: Done.

**Review by gladilov-gleb — APPROVED** [review]
Nice job

**Pull request merged**